### ISSUE TITLE
Bloc with freezed not working.

### ISSUE THREAD

**Issue by Vineeth-Kolichal** [rrousselGit/freezed#1064]
**Describe the bug**
I am using Bloc with freezed for managing states of my flutter project. But when I am trying to create a bloc today the .freezed file is not generating, I run build runner command multiple times, even though its not generating

**Comment by rrousselGit** [comment]
Please be more specific and provide a complete example.

**Comment by anugrahsputra** [comment]
I also have this issue, the .freezed.dart file is not generated for bloc/cubit. 

```bash
➜ dart run build_runner build --delete-conflicting-outputs
[INFO] Generating build script completed, took 1.0s
[INFO] Reading cached asset graph completed, took 2.3s
[INFO] Checking for updates since last build completed, took 5.1s
[WARNING] No actions completed for 16.2s, waiting on:
  - freezed on test/helper/mock.dart
  - freezed on lib/features/ayah/logic/blocs/ayahs/ayahs_bloc.dart
  - freezed on lib/features/bookmark/logics/blocs/bookmark/bookmark_bloc.dart
  - freezed on lib/features/detail_surah/logic/blocs/detail_surah/detail_surah_bloc.dart
  - freezed on lib/features/detail_surah/logic/cubits/verse_audio/verse_audio_cubit.dart
  .. and 4 more

[WARNING] No actions completed for 48.0s, waiting on:
  - freezed on test/helper/mock.dart
  - freezed on lib/features/ayah/logic/blocs/ayahs/ayahs_bloc.dart
  - freezed on lib/features/bookmark/logics/blocs/bookmark/bookmark_bloc.dart
  - freezed on lib/features/detail_surah/logic/blocs/detail_surah/detail_surah_bloc.dart
  - freezed on lib/features/detail_surah/logic/cubits/verse_audio/verse_audio_cubit.dart
  .. and 4 more

[INFO] Running build completed, took 1m 54s
[INFO] Caching finalized dependency graph completed, took 1.6s
[INFO] Succeeded after 1m 56s with 83 outputs (342 actions)
```


![image](https://github.com/rrousselGit/freezed/assets/71306482/138f6074-b434-451e-bf4f-41e0e0f19818)

**Comment by rrousselGit** [comment]
I need a complete example; something I can run.

**Comment by anugrahsputra** [comment]
not sure what you mean, but i guess to explain the issue is when creating bloc with freezed, you would do this

```dart
// state
part of 'example_bloc.dart';

@freezed
class ExampleState with _$ExampleState {
  const factory ExampleState.initial() = _Initial;
}
```
```dart
// event
part of 'example_bloc.dart';

@freezed
class ExampleEvent with _$ExampleEvent {
  const factory ExampleEvent.started() = _Started;
}
```

then you run `dart run build_runner build` to generate `example_bloc.freezed.dart`

but like @Vineeth-Kolichal said, .freezed file was not generated and that's the issue.

sorry for poor explanation

**Comment by raj457036** [comment]
Version **2.4.7** works as expected

**Comment by Dall127** [comment]
@rrousselGit , Here's a complete example of something that should technically generate a .freezed file, though is not on all the latest versions of everything. I'm not completely convinced though that this is related to Bloc (regardless of it's usage here) I can't quite pin down what is causing this, it might have even been a flutter version number bump. 

My guess is, semi confirmed by the above example as well,  is that whatever part of the build process's responsibility is for traversal, is not going through dart's  ```part 'file.dart``` ```part of 'main_file.dart'``` importing. I'm not sure if that's freezed_annotator, build_runner, freezed, or what have you. 

[lib.zip](https://github.com/rrousselGit/freezed/files/14892480/lib.zip)


pubspec 
```
name: rep_app
description: "A new Flutter project."
# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number is used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
# In Windows, build-name is used as the major, minor, and patch parts
# of the product and file versions while build-number is used as the build suffix.
version: 1.0.0+1

environment:
  sdk: '>=3.3.3 <4.0.0'

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter
  beamer: ^1.5.6
  firebase_auth: ^4.19.0
  firebase_core: ^2.28.0
  flutter_native_splash: ^2.4.0
  font_awesome_flutter: ^10.7.0
  freezed_annotation: ^2.4.1
  json_annotation: ^4.8.1
  google_fonts: ^6.2.1 
  http: ^1.2.1
  flutter_bloc: ^8.1.5
  url_launcher: ^6.2.5

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.6

dev_dependencies:
  flutter_test:
    sdk: flutter
  build_runner: ^2.4.8
  json_serializable: ^6.7.1
  freezed: ^2.4.7
  

  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^3.0.2
# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter packages.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

```

**Comment by thanhminhkma** [comment]
> Version **2.4.7** works as expected

Version 2.4.7 working properly with me too.

**Comment by Yegair** [comment]
I have encountered the same problem, I am able to reproduce it with Freezed `2.4.8` as well as `2.5.0`. Created a minimal example: https://github.com/Yegair/freezed-bloc-error-reproduction

It contains 3 Files

**main.dart**
```dart
void main(List<String> arguments) {
  final myCubit = MyCubit();

  try {
    myCubit.doSomething();
  } finally {
    myCubit.close();
  }
}
```

**cubit.dart**
```dart
import 'package:bloc/bloc.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'state.dart';
part 'cubit.freezed.dart';

class MyCubit extends Cubit<MyState> {
  MyCubit() : super(MyState.initial());

  void doSomething() {
    emit(MyState.didSomething());
  }
}
```

**state.dart**
```dart
part of 'cubit.dart';

@freezed
sealed class MyState with _$MyState {
  const MyState._();
  const factory MyState.initial() = MyStateInitial;
  const factory MyState.didSomething() = MyStateDidSomething;
}
```

I then try to generate the `cubit.freezed.dart` file as usual like

```
dart run build_runner build --delete-conflicting-outputs
```

The build runner seems to do something (there are some files produced in the `.dart_tool` directory), but the expected file `cubit.freezed.dart` is not being generated.

Reverting to `freezed: 2.4.7` reliably fixes the issue.

**Comment by specOper99** [comment]
It is happening when the @freezed annotation is inside the part file `*_event.dart` or `*_state.dart`

**Comment by AngeloAvv** [comment]
Same problem here. Reverting to 2.4.7 works properly. I don't know if it's something related to other generators, here's some of my dev deps:

``` yaml
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter
  flutter_driver:
    sdk: flutter
  auto_route_generator: ^8.0.0
  build_runner: ^2.4.9
  drift_dev: ^2.16.0
  flutter_lints: ^3.0.2
  freezed: 2.4.7
  retrofit_generator: ^8.1.0
  json_serializable: ^6.7.1
  mockito: ^5.4.4
  bloc_test: ^9.1.7
  fake_cloud_firestore: ^2.4.9
  pigeon: ^18.0.0
  data_fixture_dart: ^2.2.0
  url_launcher_platform_interface: ^2.3.2
  plugin_platform_interface: ^2.1.8
  mocktail: ^1.0.3
```

**Comment by Lzyct** [comment]
Same issue, need to revert to version 2.4.7

**Comment by YawarOsman** [comment]
Same here, and version 2.4.7 still not working

**Comment by Lzyct** [comment]
> Same here, and version 2.4.7 still not working

You should force it to use version 2.4.7 by remove `^`

Should be like this

```yaml
freezed: 2.4.7
```

**Comment by YawarOsman** [comment]
Ok, worked thanks

**Comment by TheFabbiusCorp** [comment]
Same same. Probably something at this commit broke the generator: https://github.com/rrousselGit/freezed/commit/c5377a9fc1b876fd617bc1d1469348ac109e05e7.

**Comment by lvsecoto** [comment]
Sam problem, with a freezed class in part file.

**Comment by lvsecoto** [comment]
Sam problem, with a freezed class in part file.

**Comment by lvsecoto** [comment]
Sam problem, with a freezed class in part file.

**Comment by YannMLD** [comment]
I had the same problem when I upgraded to 2.5.0 and had to go back to 2.4.7 to get Bloc working again

**Comment by PrzemyslawPluszowy** [comment]
i had the same issue, im back to 2.4.7 and still don't work

**Comment by YannMLD** [comment]
> i had the same issue, im back to 2.4.7 and still don't work
Did you follow this https://github.com/rrousselGit/freezed/issues/1064#issuecomment-2041186153 correctly ?

**Comment by PrzemyslawPluszowy** [comment]
yes,   freezed: 2.4.7
  normal class is generated, but state no

**Comment by PrzemyslawPluszowy** [comment]
ok flutter cache repair fix problem on 2.4.7

**Comment by Arenukvern** [comment]
for my case it seems like freezed ignores any "part of" files and can generate .freezed only if there is one file

what doesn't work:

models.dart have 
part 'file_1.dart';
part 'file_2.dart';
part 'models.freezed.dart';

file_1.dart have part of 'models.dart';
file_2.dart have part of 'models.dart';

if separate files to it works:

file_1.dart have only part 'file_1.freezed.dart';
file_2.dart have only part 'file_2.freezed.dart';

downgrading to 2.4.7 solves the issue

**Comment by rrousselGit** [comment]
Fixed in 2.5.1
